.github/workflows/promote-canary-to-latest.yml

```diff
@@ -0,0 +1,66 @@
+# Workflow to manually promote the returntocorp/semgrep:canary docker image
+# to returntocorp/semgrep:latest that most customers are using
+
+name: Promote returntocorp/semgrep:canary to :latest
+
+on:
+  workflow_dispatch:
+    inputs:
+      confirmed:
+        description: "Are you sure you want to do this? This is a sensitive operation"
+        type: boolean
+        required: true
+        default: false
+
+jobs:
+  promote:
+    name: promote :canary to :latest
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - env:
+          confirmed: ${{ inputs.confirmed }}
+        run: |
+          #
+          # Promoting an image via Docker tags is typically done via:
+          # docker pull source-image
+          # docker tag source-image target-image
+          # docker push target-image
+          #
+          # Sadly, this falls apart when dealing with multi-arch docker images. This is
+          # because a multi-arch image (e.g. returntocorp/semgrep:latest) doesn't actually
+          # point to a specific docker image manifest, but rather, a "manifest list" which
+          # is effectively a mapping of OS architectures to actual docker image manifests.
+          #
+          # When `docker pull` encounters a manifest list, it implicitly grabs the
+          # arch-specific image for the machine that's docker is running on, which means
+          # that the subsequent tag and push will never push the multi-arch image.
+          #
+          # Instead, we use `docker buildx imagetools create` which only operates on
+          # manifest lists. This allows us to point to a manifest list successfully
+          #
+          # For more info:
+          # - manifest lists: https://docs.docker.com/registry/spec/manifest-v2-2/
+          # - imagetools create: https://docs.docker.com/engine/reference/commandline/buildx_imagetools_create/
+
+          canary_digest=$(docker buildx imagetools inspect --raw returntocorp/semgrep:canary | jq -r .config.digest)
+          latest_digest=$(docker buildx imagetools inspect --raw returntocorp/semgrep:latest | jq -r .config.digest)
+
+          if [[ -z "${canary_digest}" ]]; then
+            echo "Error: returntocorp/semgrep:canary did not resolve to a manifest list"
+            echo "If this is urgent, you can manually run these commands to promote an arch-specific image:"
```

Comment:
actually you need the docker credentials for that, which are in the special vault I think. Could be worth adding a comment about it.